### PR TITLE
[clang] Fix incorrect inferred lifetime_capture_by attr on STL

### PR DESCRIPTION
**Pull request by hokein**
We incorrectly annotate the iterator parameter for `insert` method (`void insert(const_iterator, const value_type& value)`), because iterator is also a gsl-pointer type.

This patch fixes it.

### CODE REVIEW

File: clang/test/AST/attr-lifetime-capture-by.cpp [hunk 1]
```diff
@@ -37,67 +37,56 @@ struct vector {
...
+// CHECK-NEXT:               LifetimeCaptureByAttr {{.*}} Implicit
 
 // CHECK:       CXXMethodDecl {{.*}} insert 'void (iterator, View &&)'
-// CHECK:           ParmVarDecl {{.*}} 'iterator'
-// CHECK:               LifetimeCaptureByAttr {{.*}} Implicit
-// CHECK:           ParmVarDecl {{.*}} 'View &&'
-// CHECK:               LifetimeCaptureByAttr {{.*}} Implicit
-// CHECK-NOT:   LifetimeCaptureByAttr
+// CHECK-NEXT:           ParmVarDecl {{.*}} 'iterator'
+// CHECK-NOT:               LifetimeCaptureByAttr {{.*}} Implicit
```
Comment (usx95) [inline clang/test/AST/attr-lifetime-capture-by.cpp:51]: maybe skip this line if CHECK-NEXT is possible.
same in line 70, 87

**Review by usx95 — APPROVED** [review]
Can you also add a lit test for:
```cpp
std::vector<std::string> getVector();
std::set<std::string> s;
s.insert(getVector().begin(), getVector().end()); // FIXME: taking iterator of a temporary container without immediate dereference is almost always wrong.
```

File: clang/test/AST/attr-lifetime-capture-by.cpp (hunk 1 above)
Comment (hokein) [reply clang/test/AST/attr-lifetime-capture-by.cpp:51]: ah, right, this is even better. Done.

**Comment by hokein** [comment]
> Can you also add a lit test for:
> 
> ```c++
> std::vector<std::string> getVector();
> std::set<std::string> s;
> s.insert(getVector().begin(), getVector().end()); // FIXME: taking iterator of a temporary container without immediate dereference is almost always wrong.
> ```

I think this is a different issue, I'd prefer not adding it to this test file.

**Pull request merged**